STYLEGUIDE.md

```diff
@@ -0,0 +1,101 @@
+- 1 blank line after all headings and before and after paragraphs, directives and "|"s
+- Use Line Feed (LF, ``\n``) as the End of Line character on all platforms
+- Use UTF-8 for all files (with no BOM, as dictated by the UTF-8 standard); do not use Windows Notepad as this adds a BOM and can otherwise corrupt files easily
+- Indicate special characters (e.g. em dash, en dash, etc) by simply including them inline with Unicode
+- Suggested to avoid non-ASCII characters where minimal downside/need exists, and recommended to keep used characters within the Latin-1 set where an additional range is not required by the specific usage, or an existing proper name or other identifier uses them
+- Spaces, no tabs for indentation and otherwise
+- Natural indents (3 spaces) for directives, 4 spaces for further indents beyond that in code
+- No hard wrap, put each sentence on its own line, and try to keep them under 180 characters
+    - Wrap code blocks at a width of 70 characters from the page margin for readability.
+- Place section links above the section with one blank line between them and the section, and two blank lines between sections (as appropriate for that section level) between the link and the preceding text.
+- Always use single backticks around link titles
+- Eliminate any trailing whitespace before commiting
+- Try to keep individual files to under around ~500 lines; split into separate topics beyond that
+
+
+
+## reST Style Elements
+
+- Filenames/paths: Use :file: and ``/`` for deliminators e.g. :file:`path/to/test.py`. Example filenames should be all lowercase with no spaces or special characters unless called for by the application; real filenames should follow their actual case. Use {descriptive-name} for placeholders.
+- Keyboard Shortcuts: Use :kbd: to format, Ctrl/Alt/Shift as abbreviations (instead of Command/Option), ``-`` with no spaces to separate, and title case (including capital letters for letter keys; always include Shift explicitly if needed) e.g. :kbd:`Ctrl-Alt-Shift-P`. Use the key name or symbol as printed on a standard keyboard. For any configurable shortcut, you should include "by default" to indicate this, and must include at least one another means the command can be accessed if available (e.g. via a menu or toolbar), which should be done for all shortcuts and should include all means of executing the command, if multiple others exist.
+- GUI elements (names of things in the UI): Use ``:guilabel:`` , e.g. ``:guilabel:`Name```; make sure to use the exact name and capitalization as printed in the UI, except for the proper names of Spyder panes (i.e. those listed in the ``View > Panes`` menu), which should always be written in Title Case ("History Log", "Project Explorer", "Find in Files", etc. as they are proper names). Also, for consistency and clarity, they should be referred to as panes, rather than modules, plugins, panels, etc.; Find in Files may be called a "tool" if appropriate to the context.
```

Comment:
* There are three backticks at the end of this expression:

      ``:guilabel:`Name```

* "History Log", "Project Explorer", "Find in Files": these examples don't apply anymore. Please change them to "Variable Explorer", "IPython Console".

Reply:
```suggestion
- GUI elements (names of things in the UI): Use ``:guilabel:`` , e.g. ``:guilabel:`Name` ``; make sure to use the exact name and capitalization as printed in the UI, except for the proper names of Spyder panes (i.e. those listed in the ``View > Panes`` menu), which should always be written in Title Case ("Variable Explorer", "Code Analysis", "IPython Console", etc. as they are proper names). Also, for consistency and clarity, they should be referred to as panes, rather than modules, plugins, panels, etc.; Find in Files may be called a "tool" if appropriate to the context.
```

> There are three backticks at the end of this expression:

Yes, which is the correct number. The first is the closing backtick for `Name`, and the second two are the closing backticks for the inline verbatim text. Apparently Github's GFM parser doesn't render this correctly so I've added a space to fix that.

> Please change them to "Variable Explorer", "IPython Console".

Fixed to three up to date examples. Ironically, I just noticed that in the Spyder UI, two of those I mentioned are Sentence case, while one is Title Case, and the rest of the UI text isn't much better. This is really my fault for never doing that polish pass I was going to...I can get back to it some day.